trio/_core/_run.py

```diff
@@ -2005,6 +2014,14 @@ def unrolled_run(runner, async_fn, args, host_uses_signal_set_wakeup_fd=False):
                 if cushion < timeout:
                     timeout = cushion
                     idle_primed = True
```

Comment:
Could plausibly have this be just one variable `idle_primed` which is either `None` or one of the `IdlePrimedTypes`

Reply:
Yeah, that does look a bit nicer, now that I try it... thanks